src/contracts/evm/APTeamMultiSig/index.ts

```diff
@@ -0,0 +1,47 @@
+import { Interface } from "@ethersproject/abi";
+import { TxLog } from "types/evm";
+import abi from "./abi.json";
+
+const iface = new Interface(abi);
+
+const confirmTransactionFn = iface.getFunction("confirmTransaction");
+const executeTransactionFn = iface.getFunction("executeTransaction");
+const revokeTransactionFn = iface.getFunction("revokeTransaction");
+
+export const executionTopic = iface.encodeFilterTopics("Execution", []);
+export const executionFailureTopic = iface.encodeFilterTopics(
+  "ExecutionFailure",
+  []
+);
+
+export const confirmTransaction = {
+  encode(transactionId: number) {
+    return iface.encodeFunctionData(confirmTransactionFn, [transactionId]);
+  },
+};
+
+export const executeTransaction = {
+  encode(transactionId: number) {
+    return iface.encodeFunctionData(executeTransactionFn, [transactionId]);
+  },
+  processLogs(logs: TxLog[]) {
+    const execTopic = iface.getEventTopic("Execution");
+    const execLog = logs.find((log) => log.topics.includes(execTopic));
+    if (!!execLog) {
+      return "execution";
+    }
+    const execFailTopic = iface.getEventTopic("ExecutionFailure");
+    const execFailLog = logs.find((log) => log.topics.includes(execFailTopic));
+    if (!!execFailLog) {
+      throw new Error("ExecutionFailure");
```

Comment:
Since the current `sendTx` logic [assumes the return of a single `string | undefined` attribute](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/RC-v1.8/src/helpers/tx/sendTx/sendEVMTx.ts#L32) from the [LogProcessor](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/RC-v1.8/src/types/evm.ts#L23) (even though it can return `any` type), it really makes it hard to let the user know that an error occurred.
Since there's no other way currently to check what was the result of the transaction, implemented this to throw an error if the execution was not successful to indicate that to the user (would be caught by the calling `try/catch`).

We should definitely look into returning something else instead of [attrValue](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/RC-v1.8/src/helpers/tx/sendTx/sendEVMTx.ts#L32) for EVM transactions, as it is not always useful in its current form. Will create a proposal draft PR when I get some idea.

This is related to the discussion we had before https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1972#discussion_r1147281550

Reply:
> logic assumes the return of a single string | undefined attribute

`string | undefined` is just the current types expected to be put in place of it ( may come from `LogProcessor` or `getWasmAttribute`. We could always modify this to say `unknown`

> it really makes it hard to let the user know that an error occurred.

we only need to check the `attributeValue`  as what is currently done
```ts
const { attrValue: endowId, ...okTx } = result;
        if (!endowId) {
          return showModal(TxPrompt, {
            error: "Endowment was created but failed to save to AWS",
            tx: okTx,
          });
        }
 //or if uknown
 
if(typeof endowId !== "string") {
 //... means this data is not retrieved successfully 
 //hence show prompt to user
}

``` 

> Since there's no other way currently to check what was the result of the transaction,

there's a way to check it, and should be checked on the scope where `sendTx` was called. 


> We should definitely look into returning something else instead of attrValue for EVM transactions,

sure we could return just `EVMLog[]` but it would be in conjunction to what other senders will return say `cosmosLog[]`

and the result would need to be checked either way
```
if( cosmosLog ) { ... run cosmos parsing }
if (evmLog ) { ... run evem parsing } 
```

I think coupling the processing of this data to where it came from and just returning a general `attrValue` | `returnValue` makes more sense as we just need to check that value against our expected type for that process and show appropriate error if that expectation is not met

Reply:
@ap-justin to get the attribute from the logs, a custom log processing function needs to be created/passed from the component calling the `sendTx(type, val, log)` func. So we need to process the logs no matter how we organize this. 

IMO it would be good to make it so that if the calling component has to extract some data from the logs, they should do it as part of the `onSuccess` function/hook, where they would have access to either `EVMLog[]` (if an EVM tx) or to `CosmosLog[]` (if Cosmos tx).
Makes it easier to reason about what is going on in the code. Also allows for better type-safety (see below).

Arguments for returning logs instead of `attrValue`:
- having to pass either `attribute` or `log` parameter to `TxContent` makes the purpose of those parameters unclear unless you know how they will be used in the subsequent chain of functions (to return `TxSuccess.attrValue`)
- for someone looking at `type TxSuccess` the purpose of `TxSuccess.attrValue` is not immediately obvious and they'd have to investigate how it was populated to understand it, especially when sending "evm" Txs (one needs to pass a function named `log` into a chain of functions and in `onSuccess` they somehow get a field named `attrValue`, no apparent connection between the two)
- might not need this atm, but current setup would stop us from being able to read multiple attributes from the logs in the future. This could be addressed by returning `unknown` as you suggested, but then we lose type-safety. Can be addressed by returning logs themselves, which are typed, which the calling component can process however they see fit.

Reply:
In order not to hold the PR "hostage" to this discussion, we can treat this as something to be discussed further, so updated the `log` funcs to return `null` when the events weren't found.